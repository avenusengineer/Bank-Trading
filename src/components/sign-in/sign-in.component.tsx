import React, { useState } from "react"
import { connect } from "react-redux"

import FormInput from "../form-input/form-input.component"
import CustomButton from "../custom-button/custom-button.component"

import {
  googleSignInStart,
  emailSignInStart
} from "../../redux/user/user.actions"
import "./sign-in.styles.scss"

interface SignInProps {
  googleSignInStart: any
  emailSignInStart: any
}

const SignIn = ({ googleSignInStart, emailSignInStart }: SignInProps) => {
  const [userCreds, setCreds] = useState({
    email: "",
    password: ""
  })
  const { email, password } = userCreds

  const handleSubmit = async (event: any) => {
    event.preventDefault()
    emailSignInStart(email, password)
  }

  const handleChange = (event: any): void => {
    const { value, name } = event.target
    setCreds({ ...userCreds, [name]: value })
  }

  return (
    <div className='sign-in'>
      <h2>I already have an account</h2>
      <span>Sign in with your email and password.</span>
      <form onSubmit={handleSubmit}>
        <FormInput
          name='email'
          type='email'
          label='email'
          value={email}
          handleChange={handleChange}
          required></FormInput>
        <FormInput
          name='password'
          type='password'
          label='password'
          value={password}
          handleChange={handleChange}
          required></FormInput>
        <div className='buttons'>
          <CustomButton type='submit'>Sign In</CustomButton>
          <CustomButton
            type='button'
            onClick={googleSignInStart}
            isGoogleSignIn>
            Google Sign-In
          </CustomButton>
        </div>
      </form>
    </div>
  )
}

const mapDispatchToProps = (dispatch: Function) => ({
  googleSignInStart: () => dispatch(googleSignInStart()),
  emailSignInStart: (email: any, password: any) =>
    dispatch(emailSignInStart({ email, password }))
})

export default connect(
  null,
  mapDispatchToProps
)(SignIn)
