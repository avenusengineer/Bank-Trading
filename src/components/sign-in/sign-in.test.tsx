/* modules */
import React from "react"
import { shallow, configure, ShallowWrapper } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

/* components */
import { SignIn } from "./sign-in.component"

configure({ adapter: new Adapter() })

describe("SignIn component", () => {
  let wrapper: ShallowWrapper
  let mockGoogleSignInStart: (event: any) => void
  let mockEmailSignInStart: Function

  beforeEach(() => {
    mockGoogleSignInStart = jest.fn()
    mockEmailSignInStart = jest.fn()

    const mockProps = {
      googleSignInStart: mockGoogleSignInStart,
      emailSignInStart: mockEmailSignInStart
    }

    wrapper = shallow(<SignIn {...mockProps} />)
  })

  it("should render Signin component", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
