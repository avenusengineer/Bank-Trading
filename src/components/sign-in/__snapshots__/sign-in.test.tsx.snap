// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render Signin component 1`] = `ShallowWrapper {}`;
